librz/core/analysis_tp.c

```diff
@@ -343,21 +367,45 @@ static void type_match(RzCore *core, char *fcn_name, ut64 addr, ut64 baddr, cons
 		bool memref = false;
 		bool cmt_set = false;
 		bool res = false;
+		ssize_t cache_idx = cache_len - 1;
 		// Backtrace instruction from source sink to prev source sink
-		for (j = idx; j >= prev_idx; j--) {
+		for (j = idx; j >= prev_idx; j--, cache_idx--) {
 			ut64 instr_addr = sdb_num_get(trace, sdb_fmt("%d.addr", j), 0);
 			if (instr_addr < baddr) {
 				break;
 			}
-			RzAnalysisOp *op = rz_core_analysis_op(core, instr_addr, RZ_ANALYSIS_OP_MASK_BASIC | RZ_ANALYSIS_OP_MASK_VAL);
+			RzAnalysisOp *op = NULL;
```

Comment:
it seems to me the "get op from cache if available, otherwise call rz_core_analysis_op" can be refactored in a separate function

Reply:
Yes, this function checks last entries of the vector and sets start index.

Reply:
If analysis data for a given addr is not presented then `rz_core_analysis_op()` is called